precompile/contracts/warp/README.md

```diff
@@ -77,27 +77,24 @@ Since the predicate is encoded into the [Transaction Access List](https://eips.e
 
 Therefore, we use the [Predicate Utils](https://github.com/ava-labs/coreth/blob/master/predicate/Predicate.md) package to encode the actual byte slice of size N into the access list.
 
-### Performance Optimization: C-Chain to Subnet
+### Performance Optimization: C-Chain to Avalanche L1
 
-To support C-Chain to Subnet communication, or more generally Primary Network to Subnet communication, we special case the C-Chain for two reasons:
+For communication between the C-Chain and L1, as well as broader interactions between the Primary Network and Avalanche L1s, we've implemented special handling for the C-Chain. This special case is necessary since the Primary Network operates with the maximum number of possible validators.
 
-1. Every Subnet validator validates the C-Chain
-2. The Primary Network has the largest possible number of validators
+This large validator set in the Primary Network creates a unique challenge for Avalanche Warp Messages. Since reaching the required stake threshold would demand collecting numerous signatures, verifying messages from the Primary Network would be computationally intensive and costly. However, we've developed a more efficient solution.
```

Comment:
```suggestion
The Primary Network has a large validator set, which creates a unique challenge for Avalanche Warp Messages: Since reaching the required stake threshold would demand collecting numerous signatures, and verifying messages from the Primary Network would be computationally costly. However, we've developed a more efficient solution.
```